### DISCUSSION TITLE
Error Config Pro

### DISCUSSION THREAD

**Discussion by zTimmm** [Laravel-Backpack/community-forum#453] [General]
When I do php artisan backpack:require:pro I get an error... and I don't know how to proceed next.

**Comment by promatik** [comment]
Hi @Timmmeerrrddd!
What is the error?

**Reply by zTimmm** [reply]
![image](https://user-images.githubusercontent.com/74602195/227902503-d555fdd4-cb26-4223-91a2-07337e99a30d.png)

This is the error..

**Reply by zTimmm** [reply]
![image](https://user-images.githubusercontent.com/74602195/227906254-6e02e845-133d-4b65-87cc-b147bcfd2a9a.png)

This is what I get when I follow the steps from Manual Installation and run the composer require backpack/pro command....

**Reply by pxpm** [reply]
@Timmmeerrrddd from the error it seems that you are missing the php zip extension. 

Please follow this steps https://www.php.net/manual/en/zip.installation.php and restart your web server. 

If you have the extension already installed you should only need to enable it on the `php.ini` file and restart the web server. 

You can make sure the extension is loaded by running: `php -r "print_r(implode(', ', get_loaded_extensions()));"` and check if it's on the list. 

Cheers

**Reply by zTimmm** [reply]
@pxpm Uh I don't understand how to follow those steps honestly.... I'm new to the area of this

**Reply by pxpm** [reply]
Hey @Timmmeerrrddd no prob. I will try to help the best I can. 
Have a look at https://community.apachefriends.org/f/viewtopic.php?t=82728&p=278561#p278521

You probably have the extension installed and need to enable it. You should go to your `php.ini` configuration file (You can search on google where to find the `php.ini` file for xampp on your operating system).

You can also use: `php -i | findstr /c:"Loaded Configuration File"` to know where the configuration file is being loaded on Windows.

Hope it helps, let me know if you are facing other issues.

**Reply by zTimmm** [reply]
@pxpm It works where can I find the specific token now? I only have Username / Password

**Reply by pxpm** [reply]
It's what is needed. Go to https://backpackforlaravel.com/user/tokens while authenticated into your backpack account, and go to `Instructions` tab. You should see something similar to this: 

![image](https://user-images.githubusercontent.com/7188159/227980102-caaf2a67-e7ae-4519-b57a-06d3eac74981.png)

Cheers

**Reply by zTimmm** [reply]
@pxpm I get this..
![image](https://user-images.githubusercontent.com/74602195/227998266-2423038e-c133-49e7-a222-85010d54be78.png)

**Reply by zTimmm** [reply]
When I do composer require backpack/package-you-purchased

**Reply by pxpm** [reply]
Hey @Timmmeerrrddd you need to replace `package-you-purchased` with `pro` (the package you purchased). 
So it should be `composer require backpack/pro`.

Don't forget that if you don't have `backpack/crud` installed yet, installing `pro` will do it automatically for you, but you still need to configure it. You can do it after the composer command with `php artisan backpack:install`

Cheers

**Reply by zTimmm** [reply]
He @pxpm.. I greatly appreciate your help but what am I to do now haha

![image](https://user-images.githubusercontent.com/74602195/228088018-603599ae-4e65-4004-b630-a3697d0d80f6.png)

**Reply by pxpm** [reply]
Hey @Timmmeerrrddd, no worries, glad to help 👍 

You now should go to the link provided in the message https://gituhub.com/setting/.... (the full link) in your browser. 
There you will have instructions on how to setup the git token in your computer.

I am feeling that you are completely new to the ecosystem so I suggest to start following the https://laracasts.com/ website and watching some tutorials there. 
Some are free, some are paid. 
There is also https://laraveldaily.teachable.com/ by Povilas, that have also paid and free products.
I am not affiliated with them but I know for sure that they have great resources for people starting in Laravel. 

Backpack is built on top of Laravel, so having a basic understanding of how Laravel works will help you, as we try to follow Laravel conventions when you use Backpack software. 

Cheers

**Reply by zTimmm** [reply]
Morning @pxpm I created and entered a token... but then it says this..

![image](https://user-images.githubusercontent.com/74602195/228211417-a02d9204-01a4-4fd1-a315-c11a3aaa1d83.png)

**Reply by zTimmm** [reply]
![image](https://user-images.githubusercontent.com/74602195/228497039-291c987f-c01f-476a-960e-b6600b3fb1b9.png)

How do I solve this problem?

**Reply by pxpm** [reply]
@Timmmeerrrddd you need to add the `repositories` entry into your `composer.json`. 

If you already have `Backpack/CRUD` installed you can do `php artisan backpack:require:devtools` or you could select `devtools` as an option to install after running `php artisan backpack:install`

Cheers

**Reply by zTimmm** [reply]
Hey @pxpm, Wait maybe a very stupid question but should I have installed BackPack Pro before installation? and then php artisan backpack:install? I get a error when I do php this command: artisan backpack:require:devtools

**Reply by tabacitu** [reply]
Hey @Timmmeerrrddd - looks like you're having a lot of trouble with basic things - installation, etc. Totally understandable if you're a beginner, like you said. But as noted in our docs, Backpack is not a good choice for first-timers. You have to know modern PHP & Laravel to use Backpack, otherwise it will be waaaay to difficult to use.

If you want a refund - happy to give it to you. Reach out by email please.

Cheers!

**Reply by zTimmm** [reply]
Hey @tabacitu, The only thing I run into is the installation? 

I already created a whole admin panel... based on the free stuff? only I don't understand why it just won't link and get an error.

**Reply by pxpm** [reply]
@Timmmeerrrddd if you did with with free, you can do it with the addons too. 

I would do like this:
- install backpack/crud first : `composer require backpack/crud`
- after composer finish, I would setup the `.env` (database etc) and do `php artisan backpack:install` 

This command will give you the option to install the addons too at the end, you can select any that you bought to install. 

It should ask you for your credentials if not already setup, that you get from `backpackforlaravel.com` website and automatically create for you the `repositories` entry in your `composer.json` file. (confirm that is the case). 

Also make sure composer is using the proper github token and backpack user&token. 

You can do a `composer config --list --global` and check if you have the proper values at: 

[github-oauth.github.com] XXXXXXXXXXXXXXXXXXXXXXXXX
[http-basic.backpackforlaravel.com.username] XXXXX
[http-basic.backpackforlaravel.com.password] XXXXXXXXX

Do those steps, step by step and let me know if you have any issues. 

Cheers

**Reply by zTimmm** [reply]
I work for a company where they want to use features you offer in Pro. 

The error I get when I enter composer require backpack/crud I get this error... every time
![image](https://user-images.githubusercontent.com/74602195/229066951-aad1f354-2be1-41c9-8772-01cde2028698.png)

**Reply by pxpm** [reply]
@zTimmm you are saying you do `composer require backpack/crud` and then the error that pops is about `Devtools`. 

Can you paste here the contents of `composer.json` `require` and `require-dev` keys ? If you have `backpack/devtools` there can you remove it and try again ?

Cheers

**Reply by zTimmm** [reply]
@pxpm 

I get this..

![image](https://user-images.githubusercontent.com/74602195/229092049-5bee00d6-a396-464b-8ad0-4b073b56dc42.png)

**Reply by pxpm** [reply]
You get that when you do what ? 

You just need to open your Laravel project `composer.json` file check if there is any mention of `backpack/devtools` there, as that is what's raising the error.

Cheers

**Reply by zTimmm** [reply]
Hey, it's fixed! I think my file got corrupted.... ;D have a good Sunday guys
